### ISSUE TITLE
"[ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)" repeated and not descriptive

### ISSUE THREAD

**Issue by dnoland1** [rancher/rancher#23957]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
unknown

**Result:**
Seeing `[ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)` repeated in error logs. This log needs to be more descriptive on what network connection is timing out. The endpoint and what is being "watched" would be helpful. Having this line repeat multiple times is useless in terms of troubleshooting.

**Other details that may be helpful:**
Full log example:
```
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/10/30 09:01:54 [ERROR] watcher error unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
```
Related issue:
https://github.com/rancher/rancher/issues/23198

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
v2.3.0
- Installation option (single install/HA):
HA

**Comment by GiGurra** [comment]
Got this on all my clusters as well (4 different), all running 2.3.2.

Every 5-10 minutes I get a bunch of these logged.

**Comment by alena1108** [comment]
https://github.com/kubernetes/kubernetes/issues/6513

**Comment by GiGurra** [comment]
> [kubernetes/kubernetes#6513](https://github.com/kubernetes/kubernetes/issues/6513)

What do you mean with this reply?
I dont understand. You link to an issue from 2015 which says milestone 1.0? I now run kubernetes 1.16 using rancher 2.3.2, and this problem wasnt there in 1.14 with 2.2.8..

**Comment by liquid-cloud** [comment]
Also observing this on 2.3.2.

**Comment by huegelc** [comment]
we also notice this:
```
kubectl version
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:16:51Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:07:57Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```

Rancher 2.3.2 deployed via rke

**Comment by kinarashah** [comment]
Fix present in both master and 2.3. 

To test, make sure all watcher errors will be logged in debug mode.

**Comment by sowmyav27** [comment]
**Verified on `2.3-head` commit id: `8de4d9f2c`**

- Deployed a DO cluster, 3 nodes.
- Let the rancher server around for sometime (in this case, 10-15 minutes)

**Logs:**
```
2019/12/12 20:11:33 [DEBUG] watcher status for cluster: unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/12/12 20:11:33 [DEBUG] closing watcher for cluster
2019/12/12 20:11:33 [DEBUG] stopping watcher for cluster
2019/12/12 20:11:33 [DEBUG] watcher status for user: unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/12/12 20:11:33 [DEBUG] closing watcher for user
2019/12/12 20:11:33 [DEBUG] stopping watcher for user
2019/12/12 20:11:33 [DEBUG] watcher status for clusterAlert: unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/12/12 20:11:33 [DEBUG] closing watcher for clusterAlert
2019/12/12 20:11:33 [DEBUG] stopping watcher for clusterAlert
2019/12/12 20:11:33 [DEBUG] watcher status for projectAlertRule: unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/12/12 20:11:33 [DEBUG] closing watcher for projectAlertRule
2019/12/12 20:11:33 [DEBUG] stopping watcher for projectAlertRule
2019/12/12 20:11:33 [DEBUG] watcher status for monitorMetric: unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/12/12 20:11:33 [DEBUG] closing watcher for monitorMetric
2019/12/12 20:11:33 [DEBUG] stopping watcher for monitorMetric
2019/12/12 20:11:33 [DEBUG] watcher status for nodePool: unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/12/12 20:11:33 [DEBUG] closing watcher for nodePool
2019/12/12 20:11:33 [DEBUG] stopping watcher for nodePool
```
**Verified on `master-head` commit id: `624460445`**

- Deployed a DO cluster, 3 nodes.
- Let the rancher server around for sometime 

```
2019/12/12 20:21:50 [DEBUG] watcher status for pingConfig: unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/12/12 20:21:50 [DEBUG] closing watcher for pingConfig
2019/12/12 20:21:50 [DEBUG] stopping watcher for pingConfig
2019/12/12 20:21:50 [DEBUG] watcher status for cluster: unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/12/12 20:21:50 [DEBUG] closing watcher for cluster
2019/12/12 20:21:50 [DEBUG] stopping watcher for cluster
2019/12/12 20:21:50 [DEBUG] watcher status for rkeK8sServiceOption: unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/12/12 20:21:50 [DEBUG] closing watcher for rkeK8sServiceOption
2019/12/12 20:21:50 [DEBUG] stopping watcher for rkeK8sServiceOption
2019/12/12 20:21:50 [DEBUG] watcher status for monitorMetric: unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout exceeded while reading body)
2019/12/12 20:21:50 [DEBUG] closing watcher for monitorMetric
2019/12/12 20:21:50 [DEBUG] stopping watcher for monitorMetric
```
